sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -634,57 +619,94 @@ pi_result piPlatformsGet(pi_uint32 NumEntries, pi_platform *Platforms,
     assert(ZeDriverCount == 1);
     ZE_CALL(zeDriverGet(&ZeDriverCount, &ZeDriver));
 
-    std::lock_guard<std::mutex> Lock(PiPlatformsCacheMutex);
-    for (const pi_platform CachedPlatform : PiPlatformsCache) {
-      if (CachedPlatform->ZeDriver == ZeDriver) {
-        Platforms[0] = CachedPlatform;
-        // if the caller sent a valid NumPlatforms pointer, set it here
-        if (NumPlatforms)
-          *NumPlatforms = 1;
+    static pi_result Res = getOrCreatePlatform(ZeDriver, Platforms);
+    if (Res != PI_SUCCESS) {
+      return Res;
+    }
+  }
+
+  if (NumPlatforms)
+    *NumPlatforms = 1;
+
+  return PI_SUCCESS;
+}
+
+static pi_result getOrCreatePlatform(ze_driver_handle_t ZeDriver,
+                                     pi_platform *Platform) {
+
+  // We will retrieve the Max CommandList Cache in this lamda function so that
+  // it only has to be executed once
+  static pi_uint32 CommandListCacheSizeValue = ([]() {
```

Comment:
```suggestion
  static pi_uint32 CommandListCacheSizeValue = ([] {
```

Reply:
Done.

> > so that the cached Platforms could be
> > shared accross both piPlatformsGet and piextPlatformCreateWithNativeHandle.
> 
> I don't see the changes needed into piextPlatformCreateWithNativeHandle

Apologies, I forgot to update piextPlatformCreateWithNativeHandle with my changes. Thanks!